src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/HostDiagnosticAnalyzerProvider.cs

```diff
@@ -3,15 +3,72 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Collections.Immutable;
+using System.Reflection;
 using Microsoft.CodeAnalysis.Diagnostics;
 using Microsoft.CodeAnalysis.Workspaces.ProjectSystem;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.LanguageServer.HostWorkspace;
 internal class HostDiagnosticAnalyzerProvider : IHostDiagnosticAnalyzerProvider
 {
+    /// <summary>
+    /// The directory where the analyzers are located
+    /// </summary>
+    public string? AnalyzerDirectory { get; set; }
+
+    private const string RazorVsixExtensionId = "Microsoft.VisualStudio.RazorExtension";
```

Comment:
If we're not passing the paths directly from the client, then this (and the assembly names) should be shared with the code in https://github.com/dotnet/roslyn/pull/72482/files#diff-3dbb0a92efe7104e78e21e25b234a91e11fbe1a18f3613b3779c8eecf1febe5aR992